### PR TITLE
[REVIEW] Add rmmInitialize to nvstrings gtests

### PR DESCRIPTION
**Pull request by davidwendt**
Fix for nvstrings tests to call rmmInitialize.

### CODE REVIEW

**Comment by kkraus14** [comment]
Waiting on pushes of NVCategory and conda recipe updates and will then change label back :)

File: python/nvstrings/nvcategory.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+import weakref
+
+import rmm  # noqa: F401
```
Comment (kkraus14) [inline python/nvstrings/nvcategory.py]: This is to make RMM initialize and register a finalization on exit.

Comment (davidwendt) [reply python/nvstrings/nvcategory.py]: And the `noqa: F401` is to make `flake8` happy.

File: cpp/tests/copying/nvstring_scatter_tests.cu [hunk 2]
```diff
@@ -26,10 +26,9 @@
 #include <tests/utilities/nvcategory_utils.cuh>
 #include <tests/utilities/valid_vectors.h>
 
-template <typename T>
```
Comment (kkraus14) [inline cpp/tests/copying/nvstring_scatter_tests.cu]: @jrhemstad could use confirmation this was the correct thing to do here

File: cpp/custrings/tests/utils.h [hunk 3]
```diff
@@ -1,8 +1,25 @@
 #include <cstdio>
 #include <vector>
 
+#include <rmm/rmm.h>
+
 #include "nvstrings/NVStrings.h"
 
+#define ASSERT_RMM_SUCCEEDED(expr)  ASSERT_EQ(RMM_SUCCESS, expr)
```
Comment (kkraus14) [inline cpp/custrings/tests/utils.h]: @davidwendt review here

File: cpp/tests/copying/nvstring_scatter_tests.cu (hunk 2 above)
Comment (kkraus14) [reply cpp/tests/copying/nvstring_scatter_tests.cu]: @harrism could use a quick set of eyes here

File: cpp/custrings/tests/utils.h (hunk 3 above)
Comment (davidwendt) [reply cpp/custrings/tests/utils.h]: looks good to me

Comment (kkraus14) [reply cpp/custrings/tests/utils.h]: We should probably remove the `main` and use this fixture then

File: cpp/tests/copying/nvstring_scatter_tests.cu (hunk 2 above)
Comment (kkraus14) [reply cpp/tests/copying/nvstring_scatter_tests.cu]: There wasn't a `using...` clause so I didn't see any type parametrization which is why we removed this. We tried using `TYPED_TEST` without defining type parameters and it caused compiler errors.

**Comment by OlivierNV** [comment]
Looks like nvstrings is now ok, but custreamz is next in line for this error:
terminate called after throwing an instance of 'thrust::system::system_error'
21:30:46   what():  rmm_allocator::deallocate(): RMM_FREE: cudaErrorInitializationError: initialization error
21:30:46 Fatal Python error: Aborted

**Comment by kkraus14** [comment]
> Looks like nvstrings is now ok, but custreamz is next in line for this error:
> terminate called after throwing an instance of 'thrust::system::system_error'
> 21:30:46 what(): rmm_allocator::deallocate(): RMM_FREE: cudaErrorInitializationError: initialization error
> 21:30:46 Fatal Python error: Aborted

That's likely dependent on this: https://github.com/rapidsai/rmm/pull/157

**Comment by kkraus14** [comment]
rerun tests

**Comment by OlivierNV** [comment]
rerun tests

**Comment by harrism** [comment]
Please leave "re*run te*sts" to @kkraus14 for this PR. Can't see why the previous ones failed once you rerun.

**Comment by kkraus14** [comment]
> Please leave "re-run te-sts" to @kkraus14 for this PR. Can't see why the previous ones failed once you rerun.

I'm still keeping an eye on it. All of the GPU testers have passed so things are looking good 😄

**Comment by OlivierNV** [comment]
Are you guys waiting for something else before merging this PR ? (It's blocking all 0.11 PRs)

**Comment by kkraus14** [comment]
> Are you guys waiting for something else before merging this PR ? (It's blocking all 0.11 PRs)

It's 1:16AM EST, sorry was away from my computer... Merging now.

**Pull request merged**